cpp/oneapi/dal/algo/decision_forest/detail/model_impl.hpp

```diff
@@ -0,0 +1,40 @@
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "oneapi/dal/algo/decision_forest/common.hpp"
+
+namespace oneapi::dal::decision_forest {
+
+class detail::model_impl : public base {
+public:
+    model_impl()          = default;
+    virtual ~model_impl() = default;
+    virtual std::int64_t get_tree_count() const {
```

Comment:
Maybe
```suggestion
    virtual std::int64_t get_tree_count() const = 0;
```

Reply:
it is useful to have an ability to create an object of this class to initialize model object by default with it, without establishing special dummy empty_model class